src/Orleans.Serialization/Buffers/Writer.cs

```diff
@@ -149,9 +149,9 @@ internal Writer(TBufferWriter output, Span<byte> span, SerializerSession session
         {
             if (typeof(TBufferWriter).IsValueType)
             {
-                if (Output is IDisposable)
+                if (Output is IDisposable disposable)
```

Comment:
Could be made even more concise with:

```csharp
(Output as IDisposable)?.Dispose();
```

Reply:
I believe the existing strategy was picked for favorable JIT codegen (no boxing since the type is statically known), but I'm not 100% and don't know if either alternative produce the same output.